docs/design/diagram-types/DesignDiagramsTemplates/componentDiagrams.md

```diff
@@ -0,0 +1,76 @@
+
+ > _a component diagram depicts how components are wired together to form larger components or software systems._
+
+Component Diagrams are a type of a static structure because it focuses on the responsibility and relationships between components as part of the overall system or solution.
+
+It is not supposed to inform about the data flow, the caller or callee responsibilities, the request flows, nor any other "behavior" related characteristics.
+
+...Hold on a second... what is a Component?
+
+A [Component](https://en.wikipedia.org/wiki/Component_(UML)) is a runnable solution that performs a set of operations and can possibly be interfaced through a particular API. One can think of Components as a "stand alone" piece of software - think datastores, microservices, serverless functions, user interfaces, etc...
+
+## Essential Takeaways
+
+The primary two takeaways from a Component Diagram should be:
+
+- A quick view of all the various components (User Interface, Service, Data Storage) involved in the system
+- The immediate "touch points" that a particular Component has with other Components, including how that "touch point" is accomplished (HTTP, FTP, etc...)
+
+Depending on the complexity of the system, a team might decide to create several Component Diagrams. Where there is one diagram per Component (depicting all it's immediate "touch points" with other Components).
+
+Or if a system is simple, the team might decide to create a single Component Diagram capturing all Components in the diagram.
+
+## When to Create?
+
+Because Component Diagrams represent a high level overview of the entire system from a Component focus, it is recommended to begin the creation of this diagram from the beginning of an engagement, and update it as the various Components are identified, developed, and introduced into the system. Otherwise, if this is left till later, then there is risk that:
+
+- the team won't be able to identify areas of improvement
+- the team or other necessary stakeholders won't have a full understanding on how the system works as it is being developed
+
+Because of the inherent granularity of the system, the Component Diagrams won't have to be updated as often as [Class Diagrams](./classDiagrams.md). Things that might merit updating a Component Diagram could be:
+
+- A deletion or addition of a new Component into the system
+- A change to a system Component's interaction APIs
+- A change to a system Component's immediate "touch points" with other Components
+
+Because Component Diagrams focuses on informing the various "touch points" between Components, it requires some upfront though in order to determine what Components are needed abd what interaction mechanisms are most effective per the system requirements. This amount of upfront thought should be approached in a **pragmatic** manner - as the design may evolve over time, and that is perfectly fine,
```

Comment:
A small typo: "determine what Components are needed abd what interaction"

Reply:
Thanks for catching that @nyouens !

i’ve made that fix